nifi-pulsar-processors/src/main/java/org/apache/nifi/processors/pulsar/pubsub/ConsumePulsarRecord.java

```diff
@@ -219,7 +219,7 @@ private void consumeMessages(ProcessContext context, ProcessSession session, fin
        }
 
        RecordSchema schema = null;
-       final BlockingQueue<Message<byte[]>> parseFailures = new LinkedBlockingQueue<Message<byte[]>>();
+       final Queue<Message<byte[]>> parseFailures = new LinkedList<Message<byte[]>>();
```

Comment:
Why do you no longer use a BlockingQueue?

Reply:
Hi David -- it didn't seem to be providing any benefit, given the way the logic in consumeMessage() and handleFailures() was implemented. consumeMessages() appeared to instantiate the Queue as a local variable, and append to it in a synchronous manner. Similarly, handleFailures appeared to iterate over the queue's contents in a synchronous manner. The use of the BlockingQueue and calls to queue.poll() therefore seemed confusing and potentially unnecessary.

I've been wrong before though :-). Maybe I missed something?

Reply:
I don't recall exactly why I used it either, but I am of the "if it isn't broke, don't fix it" mentality.  Now, if the performance has improved and there are no lost messages, then I am ok with the change.

Reply:
There should be some theoretical performance improvement using the LinkedList vs. the LinkedBlockingQueue (due to the absence of locking in a situation that I contend doesn't require it anyway), but in normal operation one probably isn't going to see sufficiently frequent parse failures for it to matter.

I just pushed up a version using the LinkedBlockingQueue in place of the LinkedList.